backend/database.js

```diff
@@ -213,3 +216,101 @@ export default async () => {
   }
   await Promise.all([initVapid(), initZkpp()])
 }
+
+// Index management
+
+/**
+ * Creates a factory function that appends a value to a string index in a
+ * database.
+ * The index is identified by the provided key. The value is appended only if it
+ * does not already exist in the index.
+ *
+ * @param key - The key that identifies the index in the database.
+ * @returns A function that takes a value to append to the index.
+ */
+export const appendToIndexFactory = (key: string): (value: string) => Promise<void> => {
+  return (value: string) => {
+    // We want to ensure that the index is updated atomically (i.e., if there
+    // are multiple additions, all of them should be handled), so a queue
+    // is needed for the load & store operation.
+    return sbp('okTurtles.eventQueue/queueEvent', key, async () => {
+      // Retrieve the current index from the database using the provided key
+      const currentIndex = await sbp('chelonia.db/get', key)
+
+      // If the current index exists, check if the value is already present
+      if (currentIndex) {
+        // Check for existing value to avoid duplicates
+        if (
+          // Check if the value is at the end
+          currentIndex.endsWith('\x00' + value) ||
+          // Check if the value is at the start
+          currentIndex.startsWith(value + '\x00') ||
+          // Check if the current index is exactly the value
+          currentIndex === value
```

Comment:
How about using a normal JSON array instead of this `\x00` manipulation stuff?

Reply:
This doesn't require parsing or escaping the way JSON does.